nncf/openvino/graph/node_utils.py

```diff
@@ -468,3 +468,39 @@ def get_weighted_layer_attributes(
         }
         return LinearLayerAttributes(**kwargs)
     return GenericWeightedLayerAttributes(weight_requires_grad=False, weight_shape=attrs.get("shape", None))
+
+
+def get_activation_channel_axis(node: NNCFNode, port_id: int) -> int:
+    """
+    Returns axis number of the activation tensor which correspond to it channel.
+
+    :param node: NNCFNode instance.
+    :param port_id: Specified input port id.
+    :return: Channel axis number.
+    """
+    channel_axis = 1
+
+    if port_id > 1:
+        raise nncf.InternalError(f"{node.metatype.name} can not take more than 2 input tensors.")
+
+    if (
+        node.metatype == OVMatMulMetatype
+        and node.layer_attributes is not None
+        and node.layer_attributes.input_attributes is not None
+        and "transpose" in node.layer_attributes.input_attributes
+    ):
+        transpose = node.layer_attributes.input_attributes["transpose"]
+        channel_axis = calculate_port_based_channel_axis(port_id, transpose)
+
+    return channel_axis
```

Comment:
As I can see the main logic of this function is only used for `OVMatMulMetatype`. Also, the `port_id` is used only for `OVMatMulMetatype`. I suggest to reimplement this in a following way:
```suggestion
def get_activation_channel_axis(node: NNCFNode, port_id: Optional[int] = None) -> int:
    """
    Returns axis number of the activation tensor which correspond to it channel.

    :param node: NNCFNode instance.
    :param port_id: Specified input port id if needed.
    :return: Channel axis number.
    """
    channel_axis = 1

    if (
        node.metatype == OVMatMulMetatype
        and node.layer_attributes is not None
        and node.layer_attributes.input_attributes is not None
        and "transpose" in node.layer_attributes.input_attributes
    ):
        if port_id is None:
            raise nncf.InternalError("port_id is required for OVMatMulMetatype with transpose.")
        if port_id > 1:
            raise nncf.InternalError(f"{node.metatype.name} can not take more than 2 input tensors.")

        transpose = node.layer_attributes.input_attributes["transpose"]
        channel_axis = calculate_port_based_channel_axis(port_id, transpose)

    return channel_axis
```

Also, can we avoid these checks?
```
and node.layer_attributes is not None
and node.layer_attributes.input_attributes is not None
and "transpose" in node.layer_attributes.input_attributes
```

Reply:
This is not correct.

> is only used for OVMatMulMetatype

This is also used for the `Convolution` metatype, for example.

> Also, the port_id is used only for OVMatMulMetatype

This check also verifies `Convolution`, since it may take only 2 input ports.

> Also, can we avoid these checks?

We can't avoid it since it is also applicable for `Convolution`.